Apps/Src/FaultState.c

```diff
@@ -129,26 +143,64 @@ void EnterFaultState() {
     // TODO: create an interrupt-independent CAN interface, so we can use CAN from within a fault state
     // avoid infinite recursive faults, since our CAN Driver relies on the OS to work
     // also don't call CAN if the watchdog tripped, since CAN won't be initialized
-    /*
-    if ((Fault_BitMap & (Fault_OS | Fault_WDOG)) == 0) {
-        CANData_t data;
-        data.b = 1;
-        CANPayload_t Message;
-        Message.data = data;
-        // Push Trip message to CAN Q
-        CANbus_BlockAndSend(TRIP, Message);
-        // Push Contactor State message to CAN Q
-        data.b = 0;
-        CANbus_BlockAndSend(CONTACTOR_STATE, Message);
-    }
-    */
+
+    //Deinitialize CAN registers
+    CANbus_DeInit();
+    //Reinit CAN in fault state
+    CANbus_Init(false, true);
+    
 
 #ifdef DEBUGMODE
     char command[COMMAND_SIZE];
 #endif
     BSP_WDTimer_Init(); //This is in case we did not pass critical state and watchdog timer was not initialized
     BSP_WDTimer_Start(); 
+    uint8_t payload_idx;
```

Comment:
why do you have this variable for payload idx? Couldn't you just set all the indexes with the value they should be?

Reply:
fixed